server/api/posts/handler.go

```diff
@@ -42,3 +43,28 @@ func (handler *Handler) CreatePost(context *gin.Context) {
 
 	context.JSON(http.StatusCreated, gin.H{"message": "Post created"})
 }
+
+func (handler *Handler) GetAllPost(context *gin.Context) {
+	// tokenString := context.GetHeader("Authorization")
+	err, posts := handler.postService.GetAllPost()
+	if err != nil && posts == nil {
+		context.JSON(http.StatusInternalServerError, gin.H{"Get posts error": err})
+		return
+	}
+	context.JSON(http.StatusOK, gin.H{"POSTS\n": posts})
+}
+
+func (handler *Handler) GetPostByID(context *gin.Context) {
+	post_id := context.Param("pid")
+	pid, err := strconv.Atoi(post_id)
```

Comment:
**question**: What is this used for?

Reply:
Because the post_id is uint type so to get a specific post with given post_id , the input post_id of the url, which is the string value must change to int value and then get this value change to uint to get the post from database